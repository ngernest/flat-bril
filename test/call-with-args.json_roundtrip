{
  "functions": [
    {
      "args": [],
      "instrs": [
        {
          "dest": "x",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "dest": "y",
          "op": "const",
          "type": "int",
          "value": 2
        },
        {
          "args": [
            "x",
            "y"
          ],
          "dest": "z",
          "funcs": [
            "add2"
          ],
          "labels": [],
          "op": "call",
          "type": "int"
        },
        {
          "args": [
            "y"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        },
        {
          "args": [
            "z"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        }
      ],
      "name": "main"
    },
    {
      "args": [
        {
          "name": "x",
          "type": "int"
        },
        {
          "name": "y",
          "type": "int"
        }
      ],
      "instrs": [
        {
          "args": [
            "x",
            "y"
          ],
          "dest": "w",
          "funcs": [],
          "labels": [],
          "op": "add",
          "type": "int"
        },
        {
          "dest": "y",
          "op": "const",
          "type": "int",
          "value": 5
        },
        {
          "args": [
            "w"
          ],
          "funcs": [],
          "labels": [],
          "op": "print"
        },
        {
          "args": [
            "w"
          ],
          "funcs": [],
          "labels": [],
          "op": "ret"
        }
      ],
      "name": "add2",
      "type": "int"
    }
  ]
}
