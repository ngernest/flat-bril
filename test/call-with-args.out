@main
	op: const		dest: x 		type: int		value: 2	
	op: const		dest: y 		type: int		value: 2	
	op: call 		dest: z 		type: int		args: ["x", "y"]		funcs: ["add2"]
	op: print		args: ["y"]	
	op: print		args: ["z"]	
@add2
	op: add  		dest: w 		type: int		args: ["x", "y"]	
	op: const		dest: y 		type: int		value: 5	
	op: print		args: ["w"]	
	op: ret  		args: ["w"]	
